scripts/write-config.sh

```diff
@@ -0,0 +1,207 @@
+        usage
+        ;;
+
+    clean)
+        rm "$(pwd)/${outputdir}/*" &> /dev/null
+        exit 0
+        ;;
+
+    local)
+        k8senv=integration
+        namespace=wsmtest
+        fcenv=dev
+        ;;
+
+    dev)
+        k8senv=integration
+        namespace=dev
+        fcenv=dev
+        ;;
+
+    alpha)
+        k8senv=alpha
+        namespace=alpha
+        fcenv=alpha
+        ;;
+
+    staging)
+        k8senv=staging
+        namespace=staging
+        fcenv=staging
+        ;;
+
+
+    *) # personal env
+        k8senv=integration
+        namespace=$target
+        fcenv=dev
+        ;;
+esac
+
+# Create the output directory if it doesn't already exist
+mkdir -p "${outputdir}"
+
+# If there a config and it matches, don't regenerate
+if [ -e "${outputdir}/target.txt" ]; then
+    oldtarget=$(<"${outputdir}/target.txt")
+    if [ "$oldtarget" = "$target" ]; then
+        echo "Config for $target already written"
+        exit 0
+    fi
+fi
+
+# Read a vault path into an output file, decoding from base64
+function vaultgetb64 {
+    vaultpath=$1
+    filename=$2
+    docker run --rm -e VAULT_TOKEN="${vaulttoken}" broadinstitute/dsde-toolbox:consul-0.20.0 \
+        vault read -format=json "${vaultpath}" | jq -r .data.key | base64 -d > "${outputdir}/${filename}"
+}
+
+# Read a vault path into an output file
+function vaultget {
+    vaultpath=$1
+    filename=$2
+    docker run --rm -e VAULT_TOKEN="${vaulttoken}" broadinstitute/dsde-toolbox:consul-0.20.0 \
+        vault read -format=json "${vaultpath}" | jq -r .data > "${outputdir}/${filename}"
+}
+
+# Read database data from a vault path into a set of files
+function vaultgetdb {
+    vaultpath=$1
+    fileprefix=$2
+    fil="${outputdir}/dbtmp.json"
+    docker run --rm -e VAULT_TOKEN="${vaulttoken}" broadinstitute/dsde-toolbox:consul-0.20.0 \
+        vault read -format=json "${vaultpath}" | jq -r .data > "${fil}"
+    jq -r '.db' "${fil}" > "${outputdir}/${fileprefix}-name.txt"
+    jq -r '.password' "${fil}" > "${outputdir}/${fileprefix}-password.txt"
+    jq -r '.username' "${fil}" > "${outputdir}/${fileprefix}-username.txt"
+    rm "${fil}"
+}
+
+vaultget "secret/dsde/firecloud/${fcenv}/common/firecloud-account.json" "user-delegated-sa.json"
+vaultgetb64 "secret/dsde/terra/kernel/${k8senv}/${namespace}/workspace/app-sa" "wsm-sa.json"
+
+# The magic tokens for access to Buffer Service, Janitor, and TestRunner are in fixed places
+vaultgetb64 "secret/dsde/terra/kernel/integration/tools/crl_janitor/client-sa" "janitor-client-sa.json"
+vaultgetb64 "secret/dsde/terra/kernel/integration/tools/buffer/client-sa" "buffer-client-sa.json"
+vaultgetb64 "secret/dsde/terra/kernel/integration/common/testrunner/testrunner-sa" "testrunner-sa.json"
+
+# Test runner K8s configuration
+# TODO (PF-744): The setup for this is inconsistent. The usage is unknown. So we will skip it for now.
+#vaultgetb64 "secret/dsde/terra/kernel/integration/${namespace}/testrunner-k8s-sa" "trtmp.json"
```

Comment:
Should `integration` be `${k8senv}`?

Reply:
As far as I can see, the only testrunner-k8s-sa definitions are in integration. It looks like ichang, zloery, and wsmtest are the only places it is defined.

Do you plan to define it in other environments?